model/modeldecoder/modeldecodertest/populator.go

```diff
@@ -34,89 +34,158 @@ import (
 	"github.com/elastic/apm-server/model/modeldecoder/nullable"
 )
 
+// Values used for populating the model structs
+type Values struct {
+	Str        string
+	Int        int
+	Float      float64
+	Bool       bool
+	Time       time.Time
+	IP         net.IP
+	HTTPHeader http.Header
+	// N controls how many elements are added to a slice or a map
+	N int
+}
+
+// DefaultValues returns a Values struct initialized with non-zero values
+func DefaultValues() *Values {
+	initTime, _ := time.Parse(time.RFC3339, "2020-10-10T10:00:00Z")
+	return &Values{
+		Str:        "init",
+		Int:        1,
+		Float:      0.5,
+		Bool:       true,
+		Time:       initTime,
+		IP:         net.ParseIP("127.0.0.1"),
+		HTTPHeader: http.Header{http.CanonicalHeaderKey("user-agent"): []string{"a", "b", "c"}},
+		N:          3,
+	}
+}
+
+// UpdatedValues returns a Values struct initialized with non-zero values
```

Comment:
```suggestion
// UpdatedValues returns a Values struct initialized with non-zero values,
// different to those returned by DefaultValues.
```

(or alternatively, maybe call it "NonDefaultValues"?)